lib/spack/docs/configuration.rst

```diff
@@ -4,58 +4,136 @@
-   ``lib/spack/spack/__init__.py``
-
-By default, Spack will try to do all of its building in temporary
-space.  There are two main reasons for this.  First, Spack is designed
-to run out of a user's home directory, and on may systems the home
-directory is network mounted and potentially not a very fast
-filesystem.  We create build stages in a temporary directory to avoid
-this.  Second, many systems impose quotas on home directories, and
-``/tmp`` or similar directories often have more available space.  This
-helps conserve space for installations in users' home directories.
-
-You can customize temporary directories by editing
-``lib/spack/spack/__init__.py``.  Specifically, find this part of the file:
-
-.. code-block:: python
-
-   # Whether to build in tmp space or directly in the stage_path.
-   # If this is true, then spack will make stage directories in
-   # a tmp filesystem, and it will symlink them into stage_path.
-   use_tmp_stage = True
-
-   # Locations to use for staging and building, in order of preference
-   # Use a %u to add a username to the stage paths here, in case this
-   # is a shared filesystem.  Spack will use the first of these paths
-   # that it can create.
-   tmp_dirs = ['/nfs/tmp2/%u/spack-stage',
-               '/var/tmp/%u/spack-stage',
-               '/tmp/%u/spack-stage']
-
-The ``use_tmp_stage`` variable controls whether Spack builds
-**directly** inside the ``var/spack/`` directory.  Normally, Spack
-will try to find a temporary directory for a build, then it *symlinks*
-that temporary directory into ``var/spack/`` so that you can keep
-track of what temporary directories Spack is using.
-
-The ``tmp_dirs`` variable is a list of paths Spack should search when
-trying to find a temporary directory.  They can optionally contain a
-``%u``, which will substitute the current user's name into the path.
-The list is searched in order, and Spack will create a temporary stage
-in the first directory it finds to which it has write access.  Add
-more elements to the list to indicate where your own site's temporary
-directory is.
+.. _install-area:
+
+-----------------------------------
+Basic settings in ``config.yaml``
+-----------------------------------
+
+Spack's basic configuration options are set in ``config.yaml``.  You can
+see the default settings by looking at
+``etc/spack/defaults/config.yaml``:
+
+.. literalinclude:: ../../../etc/spack/defaults/config.yaml
+   :language: yaml
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Config file variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may notice some variables prefixed with ``$`` in the settings above.
+Spack understands several variables that can be used in values of
+configuration parameters.  They are:
+
+  * ``$spack``: path to the prefix of this spack installation
+  * ``$tempdir``: default system temporary directory (e.g, from the
+    ``$TMPDIR`` environment variable)
+  * ``$user``: name of the current user
+
```

Comment:
> So, what's the convention being flouted here, really?

The problem is, we're mixing metaphors.  As soon as we use `$`, people think Bash.  The Pythonic way to substitute named variables is with `{}`... as in `'{tmpdir}/my/path'.format(tmpdir='/tmp')`

In the past, Spack has deviated from Python convention in order to make it easier for Spack users, who presumably don't know Python.  That's why (for example) a bunch of standard Python things are imported into `Package` and renamed.  Has this design goal been dropped?  If not, I think the most consistent thing is to stick with Bash/Unix conventions --- which users are presumably more familiar with.

But I also see your point on Python's tempdir being not the same as the TMPDIR env var.  Maybe this should be justified by including or referencing the full documentation on `tempdir`, rather than just saying "(e.g, from the ``$TMPDIR`` environment variable)"